### PR TITLE
[Dist] fix etype issue in dist part pipeline

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Description

Previously, I reported a bug of edge feature support in dist part pipeline: https://github.com/dmlc/dgl/issues/4681. I just found it's not fixed yet.
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: tools/distpartitioning/utils.py [hunk 1]
```diff
@@ -377,6 +377,12 @@ def write_edge_features(edge_features, edge_file):
     edge_file     : string 
         File in which the edge information is serialized
     """
+    def format_etype(etype):
+        etype, name = etype.split('/')
+        etype = etype.split(':')[1]
+        return etype + '/' + name
+    edge_features = {format_etype(etype):
+        data for etype, data in edge_features.items()}
```
Comment (jermainewang) [inline tools/distpartitioning/utils.py:385]: Use a for loop will be cleaner than list comprehension for this case.

File: tools/distpartitioning/utils.py [hunk 2]
```diff
@@ -377,6 +377,12 @@ def write_edge_features(edge_features, edge_file):
     edge_file     : string 
         File in which the edge information is serialized
     """
+    def format_etype(etype):
+        etype, name = etype.split('/')
+        etype = etype.split(':')[1]
+        return etype + '/' + name
```
Comment (jermainewang) [inline tools/distpartitioning/utils.py:383]: I generally don't like this kind of variable shadowing. Making it difficult to read. Also, is this to convert canonical etype (in the form of `'type_1:relation:type_2'`) to etype? Is this temporary as we are shifting to canonical etype entirely?

**Review by kylasa — APPROVED** [review]
Code changes looks good to me.

File: tools/distpartitioning/utils.py (hunk 2 above)
Comment (jermainewang) [reply tools/distpartitioning/utils.py:386]: Sure. Let's fix them in one PR that completely drops the support for string etype.

**Review by jermainewang — APPROVED** [review]

**Pull request merged**